api/s3_analytics_provider.py

```diff
@@ -223,95 +203,59 @@ def collect_event(
             default=str,
             ensure_ascii=True,
         )
-        s3_uploader: S3Uploader = self._get_s3_uploader(_db)
-        analytics_file_url = s3_uploader.analytics_file_url(
-            library, license_pool, event_type, time
-        )
+        storage = self._get_storage()
+        analytics_file_key = self._get_file_key(library, license_pool, event_type, time)
 
-        # Create a temporary Representation object because S3Uploader can work only with Representation objects.
-        # NOTE: It won't be stored in the database.
-        representation = Representation(
-            media_type=MediaTypes.APPLICATION_JSON_MEDIA_TYPE, content=content
+        storage.store(
+            analytics_file_key,
+            content,
+            MediaTypes.APPLICATION_JSON_MEDIA_TYPE,
         )
-        s3_uploader.mirror_one(representation, analytics_file_url)
-
-    def _get_s3_uploader(self, db: Session) -> S3Uploader:
-        """Get an S3Uploader object associated with the provider's selected storage service.
 
-        :param db: Database session
-
-        :return: S3Uploader object associated with the provider's selected storage service
-        """
-        # To find the storage integration for the exporter, first find the
-        # external integration link associated with the provider's external
-        # integration.
-        integration_link = get_one(
-            db,
-            ExternalIntegrationLink,
-            external_integration_id=self.integration_id,
-            purpose=ExternalIntegrationLink.ANALYTICS,
-        )
-
-        if not integration_link:
-            raise CannotLoadConfiguration(
-                "The provider doesn't have an associated storage service"
-            )
-
-        # Then use the "other" integration value to find the storage integration.
-        storage_integration = get_one(
-            db, ExternalIntegration, id=integration_link.other_integration_id
+    def _get_file_key(
+        self,
+        library: Library,
+        license_pool: Optional[LicensePool],
+        event_type: str,
+        end_time: datetime.datetime,
+        start_time: Optional[datetime.datetime] = None,
+    ):
+        """The path to the analytics data file for the given library, license
+        pool and date range."""
+        root = library.short_name
+        if start_time:
+            time_part = str(start_time) + "-" + str(end_time)
+        else:
+            time_part = str(end_time)
```

Comment:
This is existing code, that was just moved around by this PR:
https://github.com/ThePalaceProject/circulation/blob/49a5d4bbb8cfca9e9e8b20f1f113b2382d0bf9c9/core/s3.py#L534-L574

Since it did not seem that determining these URLs belonged as part of the s3 class. I'd like to keep the original style intact here, so its clear the code was moved around when looking at diff / blame in the future.